### ISSUE TITLE
Accessibility of abstract stubs not preserved

### ISSUE THREAD

**Issue by mattmc3** [boo-dotnet/boo-codehaus-archive#1212]
| Field | Value |
|---|---|
| Type | Bug |
| Status | Resolved |
| Priority | Minor |
| Resolution | Fixed |
| Reporter | Cedric Vivier |
| Assignee | Cedric Vivier |
| Affects | 0.9.1 |
| Fix for | 0.9.2 |
| Components | Compiler |
| Created | 2009-07-03T13:13-0500 |
| Updated | 2009-07-07T03:28-0500 |
| Resolved | 2009-07-07T03:28-0500 |

## Description

Testcase:

```
abstract class Base:
 abstract protected def Foo():
 pass

abstract class Derived (Base):
 pass

class Concrete (Derived):
 override protected def Foo():
 pass

```

Gives:

```
BCE0090: Derived method 'Concrete.Foo' can not reduce the accessibility of 'Derived.Foo' from 'public' to 'protected'.
```

Because Derived is expanded as :

```
public abstract class Derived(Base):

 protected def constructor():
 super()

 ***public*** override def Foo() as void:
 pass

```

Accessibility should be preserved when a stub is created.

## Comments (1)

---

### Cedric Vivier, 07/Jul/09 3:28 AM

Fixed in rev. 3223

## Provenance

- <http://web.archive.org/web/20150511142112id_/https://jira.codehaus.org/browse/BOO-1212>
- <http://web.archive.org/web/20150517095328id_/https://jira.codehaus.org/browse/BOO-1212?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel>
